kvm/memory.go

```diff
@@ -43,3 +47,76 @@ func SetIdentityMapAddr(vmFd uintptr, addr uint32) error {
 
 	return err
 }
+
+type DirtyLog struct {
+	Slot    uint32
+	Padding uint32
+	BitMap  uint64
+}
+
+// GetDirtyLog provides a memory slot and return a bitmap containing any pages dirtied since the
+// last call to this ioctl. Bit 0 is the first page in the memory slot.
+// Ensure the entire structure is cleared to avoid padding issues.
+func GetDirtyLog(vmFd uintptr, dirtlog *DirtyLog) error {
+	_, err := Ioctl(vmFd,
+		IIOW(kvmGetDirtyLog, unsafe.Sizeof(DirtyLog{})),
+		uintptr(unsafe.Pointer(dirtlog)))
+
+	if errors.Is(err, syscall.ENOENT) {
+		return nil
+	}
+
+	return err
+}
+
+func SetNrMMUPages(vmFd uintptr, shadowMem uint64) error {
+	_, err := Ioctl(vmFd,
+		IIO(kvmSetNrMMUPages),
+		uintptr(shadowMem))
+
+	return err
+}
+
+type coalescedMMIOZone struct {
+	Addr   uint64
+	Size   uint32
+	PadPio uint32
+}
+
+// Coalesced I/O is a performance optimization that defers hardware register write emulation so
+// that userspace exits are avoided. It is typically used to reduce the overhead of emulating
+// frequently accessed hardware registers. When a hardware register is configured for coalesced I/O,
+// write accesses do not exit to userspace and their value is recorded in a ring buffer that is
+// shared between kernel and userspace. Coalesced I/O is used if one or more write accesses to
+// a hardware register can be deferred until a read or a write to another hardware register
+// on the same device. This last access will cause a vmexit and userspace will process accesses
+// from the ring buffer before emulating it. That will avoid exiting to userspace on repeated writes.
+// Coalesced pio is based on coalesced mmio. There is little difference between coalesced mmio and
+// pio except that coalesced pio records accesses to I/O ports.
```

Comment:
I think it is not preferable to copy the Linux documentation in its entirety. Instead, it would be appropriate to write comments in your own words or comments on the package as follows.

````go
// This package is a wrapper for the following KVM API.
// https://www.kernel.org/doc/html/latest/virt/kvm/api.html
package kvm
````

I think there are other comments that have been copied as well.

Reply:
Ok, will add comment to the package later. Removed the copied doc for now an replaced with a short comment.